CATTY-283 Convert FormulaElement extension to Swift

DhrubojyotiBis1 opened the pull request:
Converted the category defined in FormulaElement+UserVariable to Swift.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

DhrubojyotiBis1 commented:
Hi @m-herold can you tell why this swiftLint test is failing?
There is no swiftLint warning on my local machine.

Please correct me if I am doing something wrong.

m-herold commented:
It seems that there was a problem with GitHub Actions which has been resolved already.

DhrubojyotiBis1 commented:
> Thanks! I am missing tests for two execution paths: (a) only left child and (b) no child and positive test (you will have to change the _UserList_ to _UserVariable_ since we are still working with a feature toggle).

I @m-herold I have added the test for an only left child, converted the `UserList` to `UserVariable`

The test for no child was available with the name `testIsVariableUsedWithoutAnyChild` and  `testIsListUsedWithoutAnyChild` so I changed the name to `testIsVariableUsedWithNoChild` and `testIsListUsedWithNoChild`.

Please review.
